Tasks/KubernetesManifestV0/src/utils/CanaryDeploymentHelper.ts

```diff
@@ -194,10 +183,13 @@ function addCanaryLabelsAndAnnotations(inputObject: any, type: string) {
     }
 }
 
-function createCanaryObjectsArgumentString(files: string[], includeServices: boolean) {
-    const kindList = new Set();
-    const nameList = new Set();
+function addValueToList(map, key, value) {
```

Comment:
Given that the intent is to optimize the calls; you could convert the key to lowercase. 
So, when there are manifests with kind `Deployment` and `deployment` the calls would be optimized to a single one instead of two

Reply:
Okay